File: experiments/esbmc_limit_sweep/phase2_cases/04_list_parameter/good.py
def transform(xs: list[int], idx: int) -> int:
    xs[1] = xs[1] + 5
    return xs[idx]


a: int = nondet_int()
b: int = nondet_int()
c: int = nondet_int()
idx: int = nondet_int()

__ESBMC_assume(-10 <= a <= 10)
__ESBMC_assume(-10 <= b <= 10)
__ESBMC_assume(-10 <= c <= 10)
__ESBMC_assume(0 <= idx <= 2)

xs: list[int] = [a, b, c]

r: int = transform(xs, idx)

expected: int = 0

if idx == 0:
    expected = a
elif idx == 1:
    expected = b + 5
else:
    expected = c

assert r == expected

File: experiments/esbmc_limit_sweep/phase2_cases/05_string_index/good.c
#include <assert.h>

int nondet_int(void);

int classify(const char *s, int idx)
{
    if (s[idx] == 'a')
        return 1;

    if (s[idx] == 'b')
        return 2;

    return 3;
}

int main(void)
{
    int idx = nondet_int();

    __ESBMC_assume(0 <= idx && idx <= 2);

    const char s[] = "abc";

    int r = classify(s, idx);

    int expected = 3;

    if (idx == 0)
        expected = 1;
    else if (idx == 1)
        expected = 2;

    assert(r == expected);

    return 0;
}
